waspc/data/Generator/templates/server/src/auth/providers/oauth/state.ts

```diff
@@ -1,29 +1,31 @@
 import {
   Response as ExpressResponse,
   Request as ExpressRequest,
-} from "express";
-import { generateCodeVerifier, generateState } from "arctic";
+} from 'express';
+import { generateCodeVerifier, generateState } from 'arctic';
 
-import type { ProviderConfig } from "wasp/auth/providers/types";
+import type { ProviderConfig } from 'wasp/auth/providers/types';
 
-import { setOAuthCookieValue, getOAuthCookieValue } from "./cookies.js";
+import { setOAuthCookieValue, getOAuthCookieValue } from './cookies.js';
 
-export type StateType = 'state' | 'codeVerifier';
+export type RequiredStateType = 'state';
 
-export function generateAndStoreOAuthState<ST extends StateType>(
-  stateTypes: ST[],
+export type OptionalStateType = 'codeVerifier';
+
+export function generateAndStoreOAuthState<OST extends OptionalStateType>(
+  optionalStateTypes: OST[],
   provider: ProviderConfig,
-  res: ExpressResponse,
-): { [name in ST]: string } {
-  const result = {} as { [name in StateType]: string }
+  res: ExpressResponse
+): { [name in OST]: string } & { [name in RequiredStateType]: string } {
+  const result = {} as {
+    [name in OptionalStateType | RequiredStateType]: string;
+  };
 
-  if (stateTypes.includes('state' as ST)) {
-    const state = generateState();
-    setOAuthCookieValue(provider, res, 'state', state);
-    result.state = state;
-  }
+  const state = generateState();
```

Comment:
Always set the state since it's always used and will always be used for each future OAuth provider.